### ISSUE TITLE
Behavior of ggez::graphics::Canvas when DrawParam::src is changed makes me confused

### ISSUE THREAD

**Issue by atsisy** [ggez/ggez#747]
Hi. I'm developing a own game using ggez.
I found strange behavior of ggez::graphics::Canvas.

When ggez::graphics::Image is drawn with following code, drawing result will be like following.

```rust
let draw_param = ggez::graphics::DrawParam::default()
                                .src(ggez::graphics::Rect::new(0.0, 0.0, 1.0, 0.7));
ggez::graphics::draw(ctx, image_stuff, draw_param);

"Drawing Result of ggez::grapchics::Image"
---------------
|             |
|   visible   |
|-------------|
|  unvisible  |
---------------
```

But when ggez::graphics::Canvas is drawn with following code, drawing result will be like following.

```rust
let draw_param = ggez::graphics::DrawParam::default()
                                .src(ggez::graphics::Rect::new(0.0, 0.0, 1.0, 0.7));
ggez::graphics::draw(ctx, canvas_stuff, draw_param);

"Drawing Result of ggez::grapchics::Canvas"
---------------
|  unvisible  |
|-------------|
|   visible   |
|             |
---------------
```

This difference caused by flipping image when ggez::graphics::Canvas is drawn.
And I know the flipping is essential process.
But I want to draw ggez::graphics::Canvas same as ggez::graphics::Image.
So, do you know how to solve this problem?

Thank you.

**Comment by icefoxen** [comment]
That's a good question.  Fixing this sort of inconsistency is one of the main problems with `Canvas`, but it involves fixing half-done hacks in a few different places, so right now fixing it for one case will just break it somewhere else.  :-(

Now that I think of it, the *right* answer might honestly be to copy a the `Canvas` to a new texture with the same byte ordering as `Image`...  Let me look into that.

**Comment by PSteinhaus** [comment]
Hmm... I just tried fiddling with this a bit and to be honest I don't understand why we need to flip the image...
I tried how things look when we don't do it and therefore I see that it is necessary. But I don't see **why**...

Could you please try to explain that to me? I'd really like to give fixing this a try.

I guess the reason is that the ```RawRenderTargetView``` of a canvas is somehow different from the ```RawRenderTargetView``` of the screen, but... why? Can't we change that?

Btw: Not only DrawParam::src doesn't work because of this issue, but also DrawParam::rotation and DrawParam::offset...

**Comment by nobbele** [comment]
fixed by #936
